tests/test_analyzer_plugin.py

```diff
@@ -0,0 +1,165 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+
+import os
+import attr
+
+from commoncode.testcase import FileBasedTesting
+from commoncode.resource import Resource
+from commoncode.resource import build_attributes_defs
+from scancode.cli_test_utils import check_json_scan
+from scancode.cli_test_utils import run_scan_click
+
+from results_analyze.analyzer_plugin import validate_resource
+from results_analyze.analyzer_plugin import NotAnalyzableResourceException
+
+
+class AnalyzerPlugin(FileBasedTesting):
+    test_data_dir = os.path.join(os.path.dirname(__file__), "data/analyzer-plugins/")
+
+    def test_analyze_results_plugin(self):
```

Comment:
This test is failing because of a `date` resource_attribute mismatch in `check_json_scan`.

> E                 "base_name": "genshell",
> E         -       "date": "2021-01-13",
> E         ?                      ----
> E         +       "date": "2020-11-30",
> E         ?                   +++  +
> E                 "dirs_count": 0,
> E                 "extension": ".c",
> E                 "file_type": "C source, ASCII text",

I'm not sure how to resolve this. This is causing all the failed travis builds.